pkgs/os-specific/linux/msi-ec/default.nix

```
@@ -0,0 +1,51 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  linuxPackages,
+  kernel ? linuxPackages.kernel,
+}:
+stdenv.mkDerivation {
+  pname = "msi-ec-kmods";
+  version = "0-unstable-2024-09-19";
+
+  src = fetchFromGitHub {
+    owner = "m1dugh";
+    repo = "msi-ec";
+    rev = "60aad95e2c8948140232f4c4d70f4ceba821d5a5";
+    hash = "sha256-qgM4KeGoLvbVsii5YR3GTGi2MVlU+JeArzEoA+6A3a0=";
+  };
+
+  dontMakeSourcesWritable = false;
+
+  patchPhase = ''
+    cp ${./patches/Makefile} ./Makefile
+  '';
+
+  hardeningDisable = [
+    "pic"
+  ];
+
+  NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types";
+
+  makeFlags = kernel.makeFlags ++ [
+    "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+  ];
```

```suggestion
  makeFlags = kernel.makeFlags ++ [
    "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
    "INSTALL_MOD_PATH=$(out)"
    "ccflags-y+=-Wno-error=incompatible-pointer-types"
  ];
```

Setting NIX_CFLAGS_COMPILE is [usually a last resort](https://discourse.nixos.org/t/env-nix-cflags-compile-vs-cxxflags/39192/8). The Linux kernel Makefile supports ccflags-y, so we should use that instead.